### ISSUE TITLE
MultiValueDictKeyError in Wagtail 4.2

### ISSUE THREAD

**Issue by scott-schafer** [FlipperPA/wagtailcodeblock#37]
I keep getting an error in the terminal when I create a new page that is using wagtailcodeblock as a streamfield, also all of my other streamfields go missing in the page admin and the terminal shows the error listed below.

I am using wagtailcodeblock 1.28.0.0. with Wagtail 4.2 and Django 4.1.7. I installed all of the dependencies that the Wagtail doc’s recommended.

```
class HomePage(Page):

    content = StreamField([
        ('paragraph', blocks.RichTextBlock()),
        ('image', ImageChooserBlock()),
        ('code', CodeBlock())
    ], use_json_field=True)


    content_panels = Page.content_panels + [
        FieldPanel("content"),
    ]

```

This is the terminal error:

```
[02/Mar/2023 00:59:10] "GET /admin/pages/3/edit/ HTTP/1.1" 200 78981
[02/Mar/2023 00:59:10] "GET /admin/jsi18n/ HTTP/1.1" 200 3343
[02/Mar/2023 00:59:10] "GET /admin/sprite-cede3484/ HTTP/1.1" 200 108109
[02/Mar/2023 00:59:10] "GET /admin/pages/3/edit/preview/?in_preview_panel=true&mode= HTTP/1.1" 200 1844
[02/Mar/2023 00:59:10] "DELETE /admin/pages/3/edit/preview/ HTTP/1.1" 200 17
[02/Mar/2023 00:59:10] "GET /admin/sprite-cede3484/ HTTP/1.1" 200 108109
Internal Server Error: /admin/pages/3/edit/preview/
Traceback (most recent call last):
  File "/home/scott/django-sandbox/wagtailcodeblock/env/lib/python3.9/site-packages/django/utils/datastructures.py", line 84, in __getitem__
    list_ = super().__getitem__(key)
KeyError: 'content-count'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/scott/django-sandbox/wagtailcodeblock/env/lib/python3.9/site-packages/django/core/handlers/exception.py", line 56, in inner
    response = get_response(request)
  File "/home/scott/django-sandbox/wagtailcodeblock/env/lib/python3.9/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/home/scott/django-sandbox/wagtailcodeblock/env/lib/python3.9/site-packages/django/views/decorators/cache.py", line 62, in _wrapped_view_func
    response = view_func(request, *args, **kwargs)
  File "/home/scott/django-sandbox/wagtailcodeblock/env/lib/python3.9/site-packages/wagtail/admin/urls/__init__.py", line 175, in wrapper
    return view_func(request, *args, **kwargs)
  File "/home/scott/django-sandbox/wagtailcodeblock/env/lib/python3.9/site-packages/wagtail/admin/auth.py", line 182, in decorated_view
    response = view_func(request, *args, **kwargs)
  File "/home/scott/django-sandbox/wagtailcodeblock/env/lib/python3.9/site-packages/django/views/generic/base.py", line 103, in view
    return self.dispatch(request, *args, **kwargs)
  File "/home/scott/django-sandbox/wagtailcodeblock/env/lib/python3.9/site-packages/wagtail/admin/views/generic/preview.py", line 30, in dispatch
    return super().dispatch(request, *args, **kwargs)
  File "/home/scott/django-sandbox/wagtailcodeblock/env/lib/python3.9/site-packages/django/views/generic/base.py", line 142, in dispatch
    return handler(request, *args, **kwargs)
  File "/home/scott/django-sandbox/wagtailcodeblock/env/lib/python3.9/site-packages/wagtail/admin/views/generic/preview.py", line 74, in post
    is_valid = form.is_valid()
  File "/home/scott/django-sandbox/wagtailcodeblock/env/lib/python3.9/site-packages/wagtail/admin/forms/pages.py", line 176, in is_valid
    return super().is_valid()
  File "/home/scott/django-sandbox/wagtailcodeblock/env/lib/python3.9/site-packages/modelcluster/forms.py", line 343, in is_valid
    form_is_valid = super().is_valid()
  File "/home/scott/django-sandbox/wagtailcodeblock/env/lib/python3.9/site-packages/django/forms/forms.py", line 205, in is_valid
    return self.is_bound and not self.errors
  File "/home/scott/django-sandbox/wagtailcodeblock/env/lib/python3.9/site-packages/django/forms/forms.py", line 200, in errors
    self.full_clean()
  File "/home/scott/django-sandbox/wagtailcodeblock/env/lib/python3.9/site-packages/django/forms/forms.py", line 437, in full_clean
    self._clean_fields()
  File "/home/scott/django-sandbox/wagtailcodeblock/env/lib/python3.9/site-packages/django/forms/forms.py", line 444, in _clean_fields
    value = bf.initial if field.disabled else bf.data
  File "/home/scott/django-sandbox/wagtailcodeblock/env/lib/python3.9/site-packages/django/forms/boundfield.py", line 127, in data
    return self.form._widget_data_value(self.field.widget, self.html_name)
  File "/home/scott/django-sandbox/wagtailcodeblock/env/lib/python3.9/site-packages/django/forms/forms.py", line 224, in _widget_data_value
    return widget.value_from_datadict(self.data, self.files, html_name)
  File "/home/scott/django-sandbox/wagtailcodeblock/env/lib/python3.9/site-packages/wagtail/blocks/base.py", line 576, in value_from_datadict
    return self.block_def.value_from_datadict(data, files, name)
  File "/home/scott/django-sandbox/wagtailcodeblock/env/lib/python3.9/site-packages/wagtail/blocks/stream_block.py", line 104, in value_from_datadict
    count = int(data["%s-count" % prefix])
  File "/home/scott/django-sandbox/wagtailcodeblock/env/lib/python3.9/site-packages/django/utils/datastructures.py", line 86, in __getitem__
    raise MultiValueDictKeyError(key)
django.utils.datastructures.MultiValueDictKeyError: 'content-count'
[02/Mar/2023 00:59:10] "POST /admin/pages/3/edit/preview/ HTTP/1.1" 500 152270


```

**Comment by Krxplx** [comment]
I have the same problem

**Comment by FlipperPA** [comment]
We're working on Wagtail 4.2 support. I hope to have it finished by the end of the week. Apologies for the delay, we've got a few moving pieces in play.

**Comment by FlipperPA** [comment]
@scott-schafer @Krxplx I've published 1.28.0.1 to PyPI. Please give it a whirl!

https://pypi.org/project/wagtailcodeblock/1.28.0.1/
